### ISSUE TITLE
 Hello, I am very interested in this project of yours. I would like to ask if you have finished this project?

### ISSUE THREAD

**Issue by wenyishengkingkong** [winston1214/Deepsort_yolov5_ros#2]

**Comment by winston1214** [comment]
Yes, I finished this project, but there are some limitations.

**Comment by winston1214** [comment]
Yes, I have successfully completed this project, but there are some limitations. Unfortunately, I can't reveal the completed code. I'm sorry. I have submitted a paper on this and will give you a link when the paper is published! And if you have any questions about this, please feel free to ask me and I will answer them sincerely

**Comment by winston1214** [comment]
Thank you very much for your interest in my project! 😄

**Comment by wenyishengkingkong** [comment]
Thank you very much for taking the time to answer my question. I think your work is very meaningful. Can you tell me more about your code restrictions?

**Comment by wenyishengkingkong** [comment]
When I ran this code（python3 ros_track.py --source ./hd.mov --weights ./yolov5/weights/small_best.pt
）,  it -Namespace(agnostic_nms=False, augment=False, classes=[0, 1], conf_thres=0.4, config_deepsort='deep_sort_pytorch/configs/deep_sort.yaml', device='', fourcc='mp4v', img_size=640, iou_thres=0.5, output='inference/output', save_txt=False, source='./hd.mov', view_img=False, weights='./yolov5/weights/small_best.pt')
was fed back, but it seemed that there was no result or error. I don't know what to do?

**Comment by winston1214** [comment]
> When I ran this code（python3 ros_track.py --source ./hd.mov --weights ./yolov5/weights/small_best.pt
> ）, it -Namespace(agnostic_nms=False, augment=False, classes=[0, 1], conf_thres=0.4, config_deepsort='deep_sort_pytorch/configs/deep_sort.yaml', device='', fourcc='mp4v', img_size=640, iou_thres=0.5, output='inference/output', save_txt=False, source='./hd.mov', view_img=False, weights='./yolov5/weights/small_best.pt')
> was fed back, but it seemed that there was no result or error. I don't know what to do?

Is there an existing ROS on it?

This code subscribes the camera topic and proceeds with the algorithm. If you want to proceed with the reference in a video, you can follow <a href='https://github.com/mikel-brostrom/Yolov5_DeepSort_Pytorch'>the link.</a>

**Comment by winston1214** [comment]
> Thank you very much for taking the time to answer my question. I think your work is very meaningful. Can you tell me more about your code restrictions?

Once the FPS for the optical flow calculation comes out very low.

In addition, DeepSort does not classify differently by class when proceeding with multi-class classification.

I finally solved these two problems, and the code for the first problem is private and you can refer to <a href='https://github.com/winston1214/Car_Counting'>this link</a> for the second problem.

This link is an Object Counting project using DeepSort. In this project link, you see deep_sort_pytorch/deepsort/deep_sort.py and  track. py, you can solve multi-class classification problem.( I am sorry for my poor English.)


Modify code : <a href='https://github.com/winston1214/Car_Counting/blob/27bdb0c9418e434db949a49688a80a7b157f5f3b/deep_sort_pytorch/deep_sort/deep_sort.py#L30'>a correction part in deepsort.py</a>, <a href='https://github.com/winston1214/Car_Counting/blob/27bdb0c9418e434db949a49688a80a7b157f5f3b/track.py#L194'>correction part in track.py</a>

**Comment by wenyishengkingkong** [comment]
I have encountered a problem of how to use python3 and pytorch in ros. Can you help me? Thank you in advance! I wish you success in your work, happiness and health!

**Comment by winston1214** [comment]
> I have encountered a problem of how to use python3 and pytorch in ros. Can you help me? Thank you in advance! I wish you success in your work, happiness and health!

To do python version up (python2 -> python3) : <a href='https://github.com/winston1214/AICT/blob/master/ROS/Python_Upgrade.md'>python upgrade</a>

pytorch install : <a href='https://pytorch.org/get-started/locally/'>torch install</a> <- You can install it with pip3 according to your environment.

**Comment by wenyishengkingkong** [comment]
Thank you very much, and hope that you will publish a paper as soon as possible, and then let me apply your code and practice this project!

**Comment by wenyishengkingkong** [comment]
Hello, if this project is completely open source after you post the paper, can you reply? Thank you in advance!

**Comment by winston1214** [comment]
> Hello, if this project is completely open source after you post the paper, can you reply? Thank you in advance!

Sorry, this project cannot be released. But you can make it as much as you want and reveal it. To what extent do you want to implement it?

**Comment by wenyishengkingkong** [comment]
I think your idea is very good, and the code has been realized, I want to study your code idea and code seriously.

**Comment by bsljsljy** [comment]
> 我觉得你的想法很好，代码已经实现了，我想认真研究你的代码思路和代码。

大佬，你能教教我吗？你是怎么解决的，我也是这个问题。
